src/worker.cpp

```diff
@@ -105,10 +105,14 @@ Worker::onInterest(const ndn::InterestFilter&, const ndn::Interest& interest)
   }
 
   // FETCH command
-  for (const auto& delegation : interest.getForwardingHint())
-    if (delegation.name.size() > 1 && delegation.name[-1].isNumber() &&
-        delegation.name[-1].toNumber() == CommandCodes::FETCH)
+  
+for (size_t i = 0; i < interest.getForwardingHint().size(); ++i) {
+    const ndn::Name& delegation = interest.getForwardingHint()[i];
+    if (delegation.size() > 1 && delegation[-1].isNumber() &&
+        delegation[-1].toNumber() == CommandCodes::FETCH) {
       return this->fetch(interest);
+    }
+}
```

Comment:
Hmm, have you tried achieving the same with a for-each loop as well? Might be a bit cleaner.

See a small formatting nit below.

```suggestion
  
for (size_t i = 0; i < interest.getForwardingHint().size(); ++i) {
    const ndn::Name& delegation = interest.getForwardingHint()[i];
    if (delegation.size() > 1 && delegation[-1].isNumber() &&
        delegation[-1].toNumber() == CommandCodes::FETCH) {
      return this->fetch(interest);
    }
  }
```